### PR TITLE
MAISTRA-738 Add missing template files to istio-operator package

### PR DESCRIPTION
**Pull request by dgn**

### CODE REVIEW

File: istio-operator/istio-operator.spec [hunk 1]
```diff
@@ -109,12 +109,19 @@ popd
...
+
 %files
 %{_bindir}/istio-operator
 %{_sysconfdir}/istio-operator
+%{_datadir}/istio-operator
 /manifests
 
 %changelog
+* Thu Aug 08 2019 Daniel Grimm <dgrimm@redhat.com> - 1.0.0-2
+- Added templates directory
```
Comment (jwendell) [inline istio-operator/istio-operator.spec:130]: Not commenting on the PR itself, let @brian-avery do that. But if you are using -2 here, you will probably want to bump the release number on line 42.

File: istio-operator/istio-operator.spec [hunk 2]
```diff
@@ -109,12 +109,19 @@ popd
 install -d $RPM_BUILD_ROOT/manifests
 cp -ra OPERATOR/src/github.com/maistra/istio-operator/manifests-maistra/* $RPM_BUILD_ROOT/manifests
 
+# install the templates
+mkdir -p $RPM_BUILD_ROOT%{_datadir}/istio-operator/default-templates
```
Comment (brian-avery) [inline istio-operator/istio-operator.spec]:
```suggestion
TEMPLATES_DIR=$RPM_BUILD_ROOT%{_datadir}/istio-operator/default-templates
mkdir -p ${TEMPLATES_DIR}
if [[ %{community_build}  == "true" ]]; then
        cp OPERATOR/src/github.com/maistra/istio-operator/deploy/smcp-templates/maistra ${TEMPLATES_DIR}/default
else
        cp OPERATOR/src/github.com/maistra/istio-operator/deploy/smcp-templates/servicemesh ${TEMPLATES_DIR}/default
fi
cp OPERATOR/src/github.com/maistra/istio-operator/deploy/smcp-templates/base ${TEMPLATES_DIR}/default
```

Comment (brian-avery) [reply istio-operator/istio-operator.spec]: The generation of templates is in build_docker.sh which isn't executed here. This copies the relevant section over to the spec.

Comment (brian-avery) [reply istio-operator/istio-operator.spec]: Also, git_commit above should be updated to:

d6365d967cf2bc9ddd293a90bb8b3d64e34a0a0b

Comment (dgn) [reply istio-operator/istio-operator.spec]: this is the first time I see this feature used. wow

File: istio-operator/istio-operator.spec [hunk 3]
```diff
@@ -109,12 +109,26 @@ popd
...
+# install the templates
+TEMPLATES_DIR=$RPM_BUILD_ROOT%{_datadir}/istio-operator/default-templates
+mkdir -p ${TEMPLATES_DIR}
+if [[ %{community_build}  == "true" ]]; then
+        cp OPERATOR/src/github.com/maistra/istio-operator/deploy/smcp-templates/maistra ${TEMPLATES_DIR}/default
+else
+        cp OPERATOR/src/github.com/maistra/istio-operator/deploy/smcp-templates/servicemesh ${TEMPLATES_DIR}/default
+fi
+cp OPERATOR/src/github.com/maistra/istio-operator/deploy/smcp-templates/base ${TEMPLATES_DIR}/default
+cp -ra OPERATOR/src/github.com/maistra/istio-operator/tmp/_output/templates/* $RPM_BUILD_ROOT%{_datadir}/istio-operator/default-templates
```
Comment (brian-avery) [inline istio-operator/istio-operator.spec]: This line is no longer needed now.

File: istio-operator/istio-operator.spec (hunk 2 above)
Comment (dgn) [reply istio-operator/istio-operator.spec]: @brian-avery I'm not sure I understand your changes.. the if-clause I get, but the cp afterwards overwrites it again doesn't it?

File: istio-operator/istio-operator.spec (hunk 1 above)
Comment (dgn) [reply istio-operator/istio-operator.spec:130]: Done!

Comment (dgn) [reply istio-operator/istio-operator.spec:130]: Thanks :) my first spec file

**Comment by dgn** [comment]
All should be addressed

**Review by brian-avery — APPROVED** [review]

File: istio-operator/istio-operator.spec [hunk 4]
```diff
@@ -109,12 +109,27 @@ popd
 install -d $RPM_BUILD_ROOT/manifests
 cp -ra OPERATOR/src/github.com/maistra/istio-operator/manifests-maistra/* $RPM_BUILD_ROOT/manifests
 
+# install the templates
+TEMPLATES_DIR=$RPM_BUILD_ROOT%{_datadir}/istio-operator/default-templates
+mkdir -p ${TEMPLATES_DIR}
+if [[ %{community_build}  == "true" ]]; then
```
Comment (jwendell) [inline istio-operator/istio-operator.spec]: Can you surround ` %{community_build}` with quotes?

**Review by jwendell — CHANGES_REQUESTED** [review]

File: istio-operator/istio-operator.spec (hunk 4 above)
Comment (dgn) [reply istio-operator/istio-operator.spec]: Done

**Review by jwendell — APPROVED** [review]

**Review by rcernich — APPROVED** [review]

**Comment by dgn** [comment]
This now also points to `/usr/share`

**Review by jwendell — CHANGES_REQUESTED** [review]
I think we should not hard code paths in spec files. The fix yesterday in product was an emergency one. Let's discuss this later on the call.

**Comment by dgn** [comment]
> I think we should not hard code paths in spec files. The fix yesterday in product was an emergency one. Let's discuss this later on the call.

+1

**Comment by brian-avery** [comment]
I think regardless of what we do, we need a consistent path. It makes sense that it might be override-able for local dev purposes though.

**Review by jwendell — APPROVED** [review]

**Pull request merged**